executor/wasm/src/lib.rs

```diff
@@ -887,11 +892,16 @@ impl ExecutorV2 {
         block_info: BlockInfo,
         transaction_hash: TransactionHash,
         gas_limit: u64,
+        authorization_keys: BTreeSet<AccountHash>,
     ) -> Result<ExecuteResult, ExecuteError>
     where
         R: GlobalStateReader + 'static,
     {
-        let authorization_keys = BTreeSet::from_iter([initiator]);
+        let authorization_keys = if authorization_keys.is_empty() {
+            BTreeSet::from_iter([initiator])
+        } else {
+            authorization_keys
+        };
```

Comment:
That seems unnecessary - wouldn't VM1 just take authorization keys, and handle the initiator case inside?